Classes/Domain/Dto/CurrentPage.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WebVision\Deepltranslate\Core\Domain\Dto;
+
+use WebVision\Deepltranslate\Core\Exception\MissingParameterException;
+
+final class CurrentPage
+{
+    public function __construct(
+        private readonly int $uid,
+        private readonly string $title
+    ) {
+    }
+
+    /**
+     * @throws MissingParameterException
+     */
+    public static function fromArray(array $record): self
```

Comment:
Do we really need a static factory method on this "DTO" object ? Native typing is active, let's make the consuming code deal with this instead and simply use the constructor (with public CPP).